crates/validator-debt/src/worker.rs

```
@@ -898,33 +896,28 @@ async fn try_write_off_distribution_debt(
         let mut write_off_count = 0;
 
         for (leaf_index, debt) in computed_debt.debts.iter().enumerate() {
+            if rewards_distribution.checked_total_sol_debt().is_none() {
+                must_terminate_debt_write_offs = true;
+                break;
+            }
+
             if revenue_distribution::try_is_processed_leaf(processed_leaf_data, leaf_index).unwrap()
             {
                 continue;
             }
 
             let node_id = debt.node_id;
             let (deposit_key, deposit_bump) = SolanaValidatorDeposit::find_address(&node_id);
-            let deposit_account_info = wallet.connection.get_account(&deposit_key).await?;
 
             if let std::collections::hash_map::Entry::Vacant(entry) =
                 deposit_balances.entry(node_id)
             {
-                let deposit_balance = doublezero_solana_client_tools::account::balance(
-                    &deposit_account_info,
-                    &rent_sysvar,
-                );
-                entry.insert(deposit_balance);
-                tracing::debug!("Fetched deposit balance for node {node_id}: {deposit_balance}");
-            }
-
-            let deposit_balance = deposit_balances.get_mut(&node_id).unwrap();
-
-            let (_, proof) = computed_debt.find_debt_proof(&node_id).unwrap();
-
-            let wallet_key = wallet.pubkey();
+                let deposit_account_info = wallet
+                    .connection
+                    .get_account(&deposit_key)
+                    .await
+                    .unwrap_or_default();
```

Should we allow this to fail silently or bubble up the error?

---

Actually, this would default if the lamports balance is zero. I guess we could disregard trying to initialize the account in this case. We should re-evaluate this in a subsequent PR in case we do not want the accountant to be so charitable